src/balance.ts

```diff
@@ -32,6 +32,12 @@ type GetAllowance = (
   signal?: AbortSignal
 ) => Promise<Allowance | typeof NOT_FOUND_RESPONSE>;
 
+export const isAllowance = (
+  arg: Awaited<ReturnType<GetAllowance>>
```

Comment:
Instead of indirectly inferring `arg`, won't `arg: Allowance | typeof NOT_FOUND_RESPONSE` work?